src/Symfony/Component/HttpClient/Rfc9111HttpClient.php

```diff
@@ -0,0 +1,616 @@
+    {
+        if ($responses instanceof ResponseInterface) {
+            $responses = [$responses];
+        }
+
+        $mockResponses = [];
+        $asyncResponses = [];
+        $clientResponses = [];
+
+        foreach ($responses as $response) {
+            if ($response instanceof MockResponse) {
+                $mockResponses[] = $response;
+            } elseif ($response instanceof AsyncResponse) {
+                $asyncResponses[] = $response;
+            } else {
+                $clientResponses[] = $response;
+            }
+        }
+
+        if (!$mockResponses && !$clientResponses) {
+            return $this->asyncStream($asyncResponses, $timeout);
+        }
+
+        if (!$asyncResponses && !$clientResponses) {
+            return new ResponseStream(MockResponse::stream($mockResponses, $timeout));
+        }
+
+        if (!$asyncResponses && !$mockResponses) {
+            return $this->client->stream($clientResponses, $timeout);
+        }
+
+        return new ResponseStream((function () use ($mockResponses, $asyncResponses, $clientResponses, $timeout) {
+            yield from MockResponse::stream($mockResponses, $timeout);
+            yield $this->client->stream($clientResponses, $timeout);
+            yield $this->asyncStream($asyncResponses, $timeout);
+        })());
+    }
+
+    /**
+     * Returns a hash representing the request details.
+     *
+     * @param string $method the request method
+     * @param string $url    the request URL
+     *
+     * @return string the request hash
+     */
+    private function getRequestHash(string $method, string $url): string
+    {
+        return hash('xxh3', $method.$url);
+    }
+
+    /**
+     * Generates a unique metadata key based on the request hash and varying headers.
+     *
+     * @param string $requestHash       a hash representing the request details
+     * @param array  $normalizedHeaders normalized headers of the request
+     * @param array  $varyFields        headers to consider for building the variant key
+     *
+     * @return string the metadata key composed of the request hash and variant key
+     */
+    private function getMetadataKey(string $requestHash, array $normalizedHeaders, array $varyFields): string
+    {
+        $variantKey = hash(
+            'xxh3',
+            [] === $varyFields
+                ? 'NO-VARY'
+                : $this->buildVariantKey($normalizedHeaders, $varyFields)
+        );
+
+        return "metadata_{$requestHash}_{$variantKey}";
+    }
+
+    /**
+     * Build a variant key for caching, given an array of normalized headers and the vary fields.
+     *
+     * The key is a pipe-separated string of "header=value" pairs, with the special case of "header=" for headers that are not present.
+     *
+     * @param array $normalizedHeaders normalized headers
+     * @param array $varyFields        vary fields
```

Comment:
the parameter description is useless here. It only repeats parameter names with words separated by spaces instead of being camel cased.

However, we should have a precise type for both arrays:

```
@param array<string, string|string[]> $normalizedHeaders
@param string[] $varyFields
```

Reply:
Oh you accept precise types now ? Cool ! What about `list<string>` kind of types ?

Reply:
Adressed.